config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/loader/PrivateKeysDirectoryLoader.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2018 CA. All rights reserved.
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+package com.ca.apim.gateway.cagatewayconfig.tasks.zip.loader;
+
+import com.ca.apim.gateway.cagatewayconfig.tasks.zip.beans.Bundle;
+
+import javax.inject.Singleton;
+import java.io.File;
+
+@Singleton
+public class PrivateKeysDirectoryLoader implements EntityLoader {
```

Comment:
Instead of doing this look at the `com.ca.apim.gateway.cagatewayconfig.tasks.zip.loader.CertificatesLoader` and make the PrivateKeyLoader wok in a similar way.

Reply:
@kazvictor the CertificatesLoader finds and loads all the files from the directory, this would work for Private keys as well, but the second implementation when the certificate content is placed as string in the docker compose file does not work for private keys because the p12 file content can't be pasted as text.

I can keep environment part here being a directory path, then I load all the files and add them to the bundle, but I can't expect them as values of the variables in the compose file, is that ok?

Reply:
Yes that's ok, they can't be expected to come from environment variable.